services/galaxy-toolshed/galaxy-toolshed-base.js

```diff
@@ -0,0 +1,30 @@
+import Joi from 'joi'
+import { BaseJsonService } from '../index.js'
+
+const orderedInstallableRevisionsSchema = Joi.array()
+  .items(Joi.string())
+  .required()
+
+export default class BaseGalaxyToolshedService extends BaseJsonService {
+  static defaultBadgeData = { label: 'galaxy-toolshed' }
+  static baseUrl = 'https://toolshed.g2.bx.psu.edu'
+
+  async fetchOrderedInstallableRevisionsSchema({ repositoryName, owner }) {
+    return this._requestJson({
+      schema: orderedInstallableRevisionsSchema,
+      url: `${this.constructor.baseUrl}/api/repositories/get_ordered_installable_revisions?name=${repositoryName}&owner=${owner}`,
+    })
```

Comment:
Can the `repositoryName` or `owner` params ever have values that would need to be uri encoded? This should depend on what all may or may not be allowed upstream on the galaxy toolshed service

Reply:
From the documentation:
- owner: *Public names must be at least three characters in length and contain only lower-case letters, numbers, dots, underscores, and dashes ('.', '_', '-')*
- repositoryName: *should contain only [a-z0-9_-]* and *Multiple words should be separated by underscore or dashes*